### PR TITLE
Add editor-gradient-presets to get_theme_support

### LINKED ISSUE #17774: Add PHP API for registering custom gradient presents
As originally discussed in https://github.com/WordPress/gutenberg/pull/17169#issuecomment-535411779 and with work continue https://github.com/WordPress/gutenberg/pull/17603 . 

In https://github.com/WordPress/gutenberg/pull/17169 gradient presents were added to buttons. This function similar to the present colors already found in gutenberg. The present colours allow for register of list of colours in PHP using theme supports. Example. 

```
add_theme_support( 'editor-color-palette',
			array(
				array(
					'name'  => __( 'Primary', 'twentynineteen' ),
					'slug'  => 'primary',
					'color' => twentynineteen_hsl_hex( 'default' === get_theme_mod( 'primary_color' ) ? 199 : get_theme_mod( 'primary_color_hue', 199 ), 100, 33 ),
				),
				array(
					'name'  => __( 'Secondary', 'twentynineteen' ),
					'slug'  => 'secondary',
					'color' => twentynineteen_hsl_hex( 'default' === get_theme_mod( 'primary_color' ) ? 199 : get_theme_mod( 'primary_color_hue', 199 ), 100, 23 ),
				),
				array(
					'name'  => __( 'Dark Gray', 'twentynineteen' ),
					'slug'  => 'dark-gray',
					'color' => '#111',
				),
				array(
					'name'  => __( 'Light Gray', 'twentynineteen' ),
					'slug'  => 'light-gray',
					'color' => '#767676',
				),
				array(
					'name'  => __( 'White', 'twentynineteen' ),
					'slug'  => 'white',
					'color' => '#FFF',
				),
			)
);
```

A similar api should be added for gradient presents. I would recommend the name `editor-gradient-presets`.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Description
Fixes #17774

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by swissspidy** [comment]
Let's add this to the Theme Support documentation, in the same manner as the color palettes docs:

https://github.com/WordPress/gutenberg/blob/0ad0414e7a469254ef1ad105fdcf14248a986686/docs/designers-developers/developers/themes/theme-support.md#block-color-palettes

**Comment by spacedmonkey** [comment]
> Let's add this to the Theme Support documentation, in the same manner as the color palettes docs:
> 
@swissspidy I have made a pass at [docs](https://github.com/WordPress/gutenberg/pull/17841/files#diff-c14ad3a4c41ef0d7bf709f5ebe52ba64R137-R168). But I am not sure we should continue with docs, until the api is design is complete.

File: lib/experiments-page.php [hunk 1]
```diff
@@ -138,6 +138,12 @@ function gutenberg_experiments_editor_settings( $settings ) {
 		'__experimentalEnableFullSiteEditing'   => $experiments_exist ? array_key_exists( 'gutenberg-full-site-editing', get_option( 'gutenberg-experiments' ) ) : false,
 
 	);
+
+	$gradient_presets = current( (array) get_theme_support( 'editor-gradient-presets' ) );
+	if ( false !== $gradient_presets ) {
+		$experiments_settings['gradients'] = $gradient_presets;
```
Comment (jorgefilipecosta) [inline lib/experiments-page.php]: We have a problem where KSES rules in multi-sites remove spacing from gradients and then make the block look invalid.
PR fixing the problem here for default gradients here: https://github.com/WordPress/gutenberg/pull/17940

I guess we need to process the gradient values against our KSES function that removes the spaces, before setting it as setting, so even if devs pass values with space the spacing is removed.

If we prefer to solve this issue in a follow up PR, I guess we should update our samples to not include the spaces.

File: lib/experiments-page.php [hunk 2]
```diff
@@ -138,6 +138,12 @@ function gutenberg_experiments_editor_settings( $settings ) {
 		'__experimentalEnableFullSiteEditing'   => $experiments_exist ? array_key_exists( 'gutenberg-full-site-editing', get_option( 'gutenberg-experiments' ) ) : false,
 
 	);
+
+	$gradient_presets = current( (array) get_theme_support( 'editor-gradient-presets' ) );
```
Comment (jorgefilipecosta) [inline lib/experiments-page.php]: We are doing some work around these APIs, e.g: automatically generate the styles, provide better hooks/API for blocks, etc...
I guess until this work is on a good path we should keep this API experimental e.g.:
```
$gradient_presets = current( (array) get_theme_support( '__experimental-editor-gradient-presets' ) );
```

**Review by jorgefilipecosta — APPROVED** [review]
Hi @spacedmonkey, thank you for submitting this PR 👍  I left some comments, but generally, it seems to be in the right direction and can be merged when the comments are addressed.

**Comment by jorgefilipecosta** [comment]
I added a small commit that applies the suggestions. I think we can merge this PR and continue the work. If there is any comment on the last changes, we can update the things in a follow-up PR. Thank you for submitting this PR @spacedmonkey 👍

**Comment by spacedmonkey** [comment]
Thanks for the feedback. Great to see this merged!

File: docs/designers-developers/developers/themes/theme-support.md [hunk 3]
```diff
@@ -134,6 +134,46 @@ Themes are responsible for creating the classes that apply the colors in differe
 
 The class name is built appending 'has-', followed by the class name _using_ kebab case and ending with the context name.
 
+### Block Gradient Presents
+
+Different blocks have the possibility of selecting from a list of predined of gradients. The block editor provides a default gradient presets, but a theme can overwrite them and provide its own:
+
+```php
+add_theme_support(
+	'__experimental-editor-gradient-presets',
```
Comment (youknowriad) [inline docs/designers-developers/developers/themes/theme-support.md]: Any particular reason to keep this as experimental for now? When do you think we should make it stable?

**Pull request merged**